src/flamenco/vm/syscall/fd_vm_syscall_cpi_common.c

```diff
@@ -753,11 +753,11 @@ VM_SYSCALL_CPI_ENTRYPOINT( void *  _vm,
   caller_lamports_h = 0UL;
   caller_lamports_l = 0UL;
   err = fd_instr_info_sum_account_lamports( vm->instr_ctx->instr, &caller_lamports_h, &caller_lamports_l );
-  if ( FD_UNLIKELY( err ) ) return FD_VM_ERR_INSTR_ERR;
+  if ( FD_UNLIKELY( err ) ) return err;
 
   if( caller_lamports_h != vm->instr_ctx->instr->starting_lamports_h || 
       caller_lamports_l != vm->instr_ctx->instr->starting_lamports_l ) {
-    return FD_VM_ERR_INSTR_ERR;
+    return FD_VM_ERR_SYSCALL_CPI_LAMPORTS_MISMATCH;
   }
 
   return FD_VM_SUCCESS;
```

Comment:
Can remove this check now https://github.com/firedancer-io/firedancer/pull/3049 is merged